apis/apps/v1/componentdefinition_types.go

```diff
@@ -0,0 +1,1973 @@
+	//   is being cleaned up.
+	//
+	// Note: This field is immutable once it has been set.
+	//
+	// +optional
+	PreTerminate *Action `json:"preTerminate,omitempty"`
+
+	// Defines the procedure which is invoked regularly to assess the role of replicas.
+	//
+	// This action is periodically triggered by Lorry at the specified interval to determine the role of each replica.
+	// Upon successful execution, the action's output designates the role of the replica,
+	// which should match one of the predefined role names within `componentDefinition.spec.roles`.
+	// The output is then compared with the previous successful execution result.
+	// If a role change is detected, an event is generated to inform the controller,
+	// which initiates an update of the replica's role.
+	//
+	// Defining a RoleProbe Action for a Component is required if roles are defined for the Component.
+	// It ensures replicas are correctly labeled with their respective roles.
+	// Without this, services that rely on roleSelectors might improperly direct traffic to wrong replicas.
+	//
+	// The container executing this action has access to following environment variables:
+	//
+	// - KB_POD_FQDN: The FQDN of the Pod whose role is being assessed.
+	// - KB_SERVICE_PORT: The port used by the database service.
+	// - KB_SERVICE_USER: The username with the necessary permissions to interact with the database service.
+	// - KB_SERVICE_PASSWORD: The corresponding password for KB_SERVICE_USER to authenticate with the database service.
+	//
+	// Expected output of this action:
+	// - On Success: The determined role of the replica, which must align with one of the roles specified
+	//   in the component definition.
+	// - On Failure: An error message, if applicable, indicating why the action failed.
+	//
+	// Note: This field is immutable once it has been set.
+	//
+	// +optional
+	RoleProbe *Probe `json:"roleProbe,omitempty"`
+
+	// Defines the procedure for a controlled transition of leadership from the current leader to a new replica.
+	// This approach aims to minimize downtime and maintain availability in systems with a leader-follower topology,
+	// during events such as planned maintenance or when performing stop, shutdown, restart, or upgrade operations
+	// involving the current leader node.
+	//
+	// The container executing this action has access to following environment variables:
+	//
+	// - KB_SWITCHOVER_CANDIDATE_NAME: The name of the pod for the new leader candidate, which may not be specified (empty).
+	// - KB_SWITCHOVER_CANDIDATE_FQDN: The FQDN of the new leader candidate's pod, which may not be specified (empty).
+	// - KB_LEADER_POD_IP: The IP address of the current leader's pod prior to the switchover.
+	// - KB_LEADER_POD_NAME: The name of the current leader's pod prior to the switchover.
+	// - KB_LEADER_POD_FQDN: The FQDN of the current leader's pod prior to the switchover.
```

Comment:
Switchover refers to transferring the current pod's role to another pod candidate with a different role. 

The parameters should include the roles of the current pod and the candidate pod(s).